SystemDocumentation/SystemDocs.txt

```diff
@@ -0,0 +1,206 @@
+users cart.
+
+
+@app.route("/checkout")
+This method leads the cashier from the cart to the checkout page. This route renders the checkout page
+
+@app.route("/methodOfPayment")
+Here we have the last method for the cashier to hit, this method will accept a form of payment for the items on the checkout page. After
+the cashier has entered payment it will clear the cart and the total for the current person.
+
+@app.route("/search")
+This method will take input from the user and search the database for what the user is looking by
+Possible search queries are:
+price
+product_id
+product name
+supplier
+It will return what the database finds in the table on the POS page
+
+
+@app.route("/profitReport")
+This method does pretty much what it says. It gives back a profit report of all the items in the inventory.
+
+@app.route("/managerSearch")
+This does the same thing as before, but it searches the database for the manager.
+
+@app.route("/createSale")
+This will create sales and lower or give price decreases on products in the database. Only the manager can access this functionality.
+
+
+UserLoginPackage.py -
+Overview: This code allows us to make users and create session objects for those users and allow them to log in and use the system.
+
+The login html is used to when requireLogin is called
+
+loginErrorHtml is thrown when someone enters bad credentials
+
+Use Cases/Functions:
+requireLogin: This function can be called to make someone log in before accessing parts of the system, keeping it secure so not just anyone can access it
+
+requireManagerLogin: This function can be called where a manager is supposed to log in and it makes sure a manager is logging in not just a cashier.
+
+logout: This function logs both managers and cashiers out, it clears all session objects associated with them as well.
+
+login: This function is used to log a user in and it will check the username and the password to make sure they are both correct and in the db, it will then
+create new session object associated with that user so we can have multiple users going at the same time not just one person using it.
+
+
+
+script.py -
+Overview: This code allows us to run tests before deploying our production environment. It makes sure there are no errors in anything we have coded before giving it to the
+customer.
+
+Use Cases:
+
+getCreds, this function will get the credentials from a file to run the tests with.
+
+client, this will send a request to our site to make sure its up and working
+
+test_client, makes sure the log in page pops up when we access the site.
+
+test_inventory, makes sure the inventory endpoint gives something back when we hit it.
+
+test_pos, hits the pos page to make sure its up and running.
+
+test_logout, makes sure the logout endpoint is up and running.
+
+
+
+Database Information(Data) - Our database is also hosted on heroku, it is accessed through our code using heroku toolbelt. 
```

Comment:
We actually use the `psycopg2` module to connect to the db in the code. The toolbelt is the terminal interface that we use to manually look in the db